activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -771,11 +771,7 @@ def index_name(table_name, options) #:nodoc:
       end
 
       # Verifies the existence of an index with a given name.
-      #
-      # The default argument is returned if the underlying implementation does not define the indexes method,
-      # as there's no way to determine the correct answer in that case.
-      def index_name_exists?(table_name, index_name, default)
-        return default unless respond_to?(:indexes)
+      def index_name_exists?(table_name, index_name, default = nil)
```

Comment:
Does anyone actually use this method externally? When [I added the `index_exists?` (and `column_exists?`)][1] methods in 11ff3da I renamed this method from `index_exists?` because it seemed to only be used internally at the time and was added as part of a general refactoring of `add_index` and `remove_index` in 99bcce7. I was wandering whether the whole method should be deprecated and made private?

[1]: https://rails.lighthouseapp.com/projects/8994/tickets/4219